woocommerce/woocommerce-ios
# [Tablet Orders] Update completion button based on flag, and navigation title based on size class

Closes: #12057

## Description
This PR addresses 2 design tweaks:
* On iPhones, portrait: The product selector completion button currently displays the number of selected products when we select or deselect them. With the new designs this will only show either "Done" if no products have been selected, or "Add Products" text if any has been selected.
* On iPhones, portrait: The "selected products" text that was showing in the completion button, now has been moved to the navigation header.

<img width="282" alt="Screenshot 2024-02-28 at 11 08 40" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/120993da-9f37-4f12-b744-f1da26591590">

## Testing instructions
* With `.sideBySideViewForOrderForm` disabled, observe that the view behaves as before: "Add Product" shows in the navigation bar, and the button is updated with the amount of selected products.
* With `.sideBySideViewForOrderForm` enabled, observe that the view behaves as after: The navigation bar shows the number of selected products, and the button only switches between "Done" (if no products have been selected) or "Add Products" (if any has been selected).

| iPhone before | iPhone after |
|--------|--------|
| ![simulator_screenshot_1EBEB86D-5A87-4031-87A5-0EEEDD373097](https://github.com/woocommerce/woocommerce-ios/assets/3812076/46a6b8e5-ed84-4cf0-a31c-356cf96b28ec) | ![simulator_screenshot_26EEC54D-6CFA-436F-A28C-B78A2A3E3F09](https://github.com/woocommerce/woocommerce-ios/assets/3812076/0d6d3732-7a86-4697-a255-345b98d726ba) | 

## Edge case in tablets:

There's one edge case that is not addressed in this PR: If the feature flag is enabled, and we run the app using split view where the product selector is narrow, but not narrow enough to use modal-on-modal, then we'll see only "Add Products" in the navigation title. This will not happen if we "move to that view", only when we initialize the product selector in that specific setup.

![20240228 edge case tablet splitview narrow](https://github.com/woocommerce/woocommerce-ios/assets/3812076/713bd2b3-497a-4b1a-a92d-65dce4edb128)

This is because the `onChange` property that checks the current subview width and needs to notify the parent in order to update the navigation title is not called on initialization, but only when the property changes. Addressing this seems to be a better fit when we attempt to switch from the current usage of GeometryReader to specific adaptive views.
